lib/phoenix_live_view.ex

```diff
@@ -740,9 +740,45 @@ defmodule Phoenix.LiveView do
   also uses `assign_new/3` to fetch the `:current_user` in its `mount/3` callback,
   as above, the assign will be fetched from the parent LiveView, once again
   avoiding additional database queries.
+
+  Note that `fun` also provides access to the previously assigned values:
+
+      assigns =
+          assigns
+          |> assign_new(:foo, fn -> "foo" end)
+          |> assign_new(:bar, fn %{foo: foo} -> foo <> "bar" end)
   '''
   def assign_new(socket_or_assigns, key, fun)
 
+  def assign_new(%Socket{} = socket, key, fun) when is_function(fun, 1) do
+    validate_assign_key!(key)
+
+    case socket do
+      %{assigns: %{^key => _}} ->
+        socket
+
+      %{private: %{assign_new: {assigns, keys}}} ->
+        # It is important to store the keys even if they are not in assigns
+        # because maybe the controller doesn't have it but the view does.
+        socket = put_in(socket.private.assign_new, {assigns, [key | keys]})
+
+        # take only public assigns from `socket.assigns`
+        new_assigns =
```

Comment:
Another solution to clean up `socket.assigns` is to call https://hexdocs.pm/phoenix_live_view/Phoenix.LiveView.Helpers.html#assigns_to_attributes/2, see https://github.com/phoenixframework/phoenix_live_view/pull/1999#discussion_r862013992

I forgot about that function but seems like a good approach. WDYT?

Reply:
I'm not so sure thats better. At the moment I'm seeing that I have to provide a list of ignores which means I have to figure out which to ignore first.  I gave it a spin and also learned it returns a Keyword list which also means I have to manage that when merging the two.  Did you have a specific thought in mind?  its not inherently coming to me as I'm trying it.

Reply:
Never mind, I was thinking about reusing that function but it won't work.